src/common/format/pretty-print.js

```diff
@@ -0,0 +1,56 @@
+import prettier from 'prettier/standalone';
+import markdown from 'prettier/parser-markdown';
+import unindent from 'strip-indent';
+
+const widths = { subject: 50, body: 72 };
+
+const config = { parser: 'markdown', plugins: [markdown] };
+
+function format(text, width) {
+  const options = { ...config, printWidth: width, proseWrap: 'always' };
+  return prettier.format(text, options);
+}
+
+function splitf(string, separator) {
+  const position = string.indexOf(separator);
+
+  if (position < 0) return [string];
+
+  const head = string.substring(0, position);
+  const tail = string.substring(position + separator.length);
+
+  return [head, tail];
+}
+
+function capitalizef(string) {
+  return string.replace(/[a-zA-Z]/, w => w.toUpperCase());
+}
+
+function gitsubject(string) {
+  const subject = capitalizef(string.trim());
+
+  if (subject.length > widths.subject) {
```

Comment:
Interesting idea 👍 . While probably way too radical for e.g. a linter / autoformatter for commit messages, I feel like this is rather inspiring / encouraging here. Like it!

Reply:
I am not sure, but I think there might be linters/linter plugins in projects like danger etc.

I didn't find a formatter though which is why I wrote this thing and – like you – I was suprised it didn't take a whole lot of code thanks to projects like prettier 🙌

Reply:
…we could release it as a separate `pretty-print-git-message` package though (the name is completely a placeholder) if you think that's useful?

Reply:
🎉 nice idea! Not sure about the extra maintenance effort though. Of couse, if everythink works fine, we rarely have to touch it. But if some leftpad party or anything else happens, it's one more thing to take care of...